torchsynth/synth.py

```diff
@@ -272,19 +272,6 @@ def randomize(self, seed: Optional[int] = None):
         Randomize all parameters
         """
         parameters = [param for _, param in sorted(self.named_parameters())]
-
-        # https://github.com/torchsynth/torchsynth/issues/253
-        if (
-            self.batch_size != BATCH_SIZE_FOR_REPRODUCIBILITY
-            and self.synthconfig.reproducible
-        ):
-            raise ValueError(
-                "Reproducibility currently only supported "
-                f"with batch_size = {BATCH_SIZE_FOR_REPRODUCIBILITY}. "
-                "If you want a different batch_size, "
-                "initialize SynthConfig with reproducible=False"
-            )
-
```

Comment:
The code below already supports any batch size.

Reply:
Yeah looks good. Have you profiled? Does this PR affect speed at all

Reply:
The code is already in main and is included in the profiles we did for the paper :)

Reply:
i.e. the solution we came up with a week ago or whatever was deterministic for different batch sizes and is also fast.